src/users/user.php

```diff
@@ -0,0 +1,367 @@
+<?php
+/**
+ * @package     Com.Api
+ * @subpackage  users
+ * @copyright   Copyright (C) 2009-2017 Techjoomla, Techjoomla Pvt. Ltd. All rights reserved.
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+// No direct access.
+defined('_JEXEC') or die();
+
+/**
+ * User Api.
+ * Creates a new user, updates an existing user and gets data of an user
+ *
+ * @package  Com.Api
+ *
+ * @since    2.0
+ */
+class UsersApiResourceUser extends ApiResource
+{
+	/**
+	 * Function to create and edit user record.
+	 *
+	 * @return object|void User details on success. raise error on failure.
+	 *
+	 * @since   2.0
+	 */
+	public function post()
+	{
+		$app = JFactory::getApplication();
+		$userIdentifier = $app->input->get('id', 0, 'String');
+		$formData = array();
+		$formData['username'] = $app->input->get('username', 0, 'String');
```

Comment:
Can't this all be done with a single `getArray()` ?

Reply:
Done